client/provider.go

```diff
@@ -205,3 +208,44 @@ func (o *repositoryKeyOption) apply(cfg *providerConfig) {
 func (o *repositoryKeyOption) String() string {
 	return fmt.Sprintf("%+v", *o)
 }
+
+type otelTracingOption struct{}
+
+func WithOtelTracing() ProviderOption {
+	return &otelTracingOption{}
+}
+
+func (o *otelTracingOption) apply(cfg *providerConfig) {
+	cfg.otelTracing = true
+}
+
+func (o *otelTracingOption) String() string {
+	return "otelTracing"
+}
+
+type otelTracing struct{}
+
+func (o *otelTracing) addTracingEvent(ctx context.Context, key, stringValue, version string) {
+	if o == nil {
+		return
+	}
+
+	span := trace.SpanFromContext(ctx)
+
+	var attributes []attribute.KeyValue
+	if len(stringValue) > 0 {
+		attributes = append(attributes, attribute.String(fmt.Sprintf("config.%s.string_value", key), stringValue))
```

Comment:
Bit unclear on the attribute naming choices, is it intentional to have both `config.xxx....` and `lekko.xxx`?

Reply:
Good catch, I forgot to update that. Agree, it should be consistent.